runtime/src/main/java/org/corfudb/runtime/object/StreamViewSMRAdapter.java

```diff
@@ -96,6 +97,10 @@ public void gc(long trimMark) {
      */
     public List<SMREntry> current() {
         ILogData data = streamView.current();
+        if (data != null && data.isHole()) {
+            throw new HoleInStreamException();
+        }
+
```

Comment:
Rather than signaling this condition via an exception, maybe a better approach would be to have a globally unique instance that signals a hole.

Reply:
We changed to put hole into ResolvedQueue, and return an empty list<SMREntry> to be ignored by VLO.